include/MixHelpers.h

```diff
@@ -55,19 +56,19 @@ void addMultiplied( sampleFrame* dst, const sampleFrame* src, float coeffSrc, in
 void addSwappedMultiplied( sampleFrame* dst, const sampleFrame* src, float coeffSrc, int frames );
 
 /*! \brief Add samples from src multiplied by coeffSrc and coeffSrcBuf to dst */
-void addMultipliedByBuffer( sampleFrame* dst, const sampleFrame* src, float coeffSrc, ValueBuffer * coeffSrcBuf, int frames );
+void addMultipliedByBuffer( sampleFrame* dst, const sampleFrame* src, float coeffSrc, std::vector<float> * coeffSrcBuf, int frames );
```

Comment:
I believe this is what @sakertooth mentioned. So what's the best way to handle this?

Reply:
The reason why I mentioned not to take a `std::vector<float>` by pointer is because these functions do not need to know what container is being used. They can simply get a range of floats and work not only just as well, but also allow for other containers besides vectors to be used. Like @DomClark  said, we were already passing a vector of floats by pointer, but we can improve this code further (since we are removing `ValueBuffer` anyway) and make the API more flexible to allow for anything that represents an array of floats (it could work with `std::vector`, `std::array`, a regular C-array, pretty much anything that can act as an array of floats).

What I would do is pass in a `float*` argument that points to an array of floats. There might not be a need to pass in a size for the array since it seems like the buffers passed into this function use `frames` as their size. Ideally, we would use [`std::span`](https://en.cppreference.com/w/cpp/container/span) instead of raw pointers, but AFAIK we don't have support for it yet.

Reply:
Okk so basically, this will turn into kind of a C style implementation if what I'm thinking of is right.

Reply:
For example, this:
```cpp
void addMultipliedByBuffer( sampleFrame* dst, const sampleFrame* src, float coeffSrc, std::vector<float> * coeffSrcBuf, int frames );
```

Will turn into this:
```cpp
void addMultipliedByBuffer(sampleFrame* dst, const sampleFrame* src, float coeffSrc, float* coeffSrcBuf, int frames);
```
This works with any container of floats, not just with `std::vector`.

Reply:
Let me know if you want some help with this, since I believe you will also have to make changes elsewhere at the call sites and make sure they pass in the right argument.

Reply:
I have converted stuff to use `float*` directly. Haven't tested for regressions tho.

> Just to check, is frames the number of floats in coeffSrcBuf or will we need to add a new parameter?

I believe we might need to. The function seems to be called in Mixer.cpp line 143 and co

Reply:
I did something, The CI will tell the rest.

Reply:
Need some assistance here, looks like i bit more than i can chew over here.

Reply:
or did I overcomplicate it?

Reply:
Check your `std::clamp`s, missing commas

Reply:
> Check your std::clamps, missing commas

I did miss a , in one line and copy paste the mistake, so i missed the , everywhere.